### PR TITLE
Adding initial configuration of the gatsby theme

### LINKED ISSUE #3: Config gatsby theme
**Is your feature request related to a problem? Please describe.**
The andromedocs will be a gatsby theme that the user can install and use.

**Describe the solution you'd like**
Something like [this](https://www.gatsbyjs.com/tutorial/building-a-theme/)

Also include: 

- MDX plugin
- Route/path fronttmatter attribute

### PR DESCRIPTION
**Pull request by JGuilhermeCoelho**
#### What is the purpose of this pull request?
Add the /theme directory with the initial gatsby theme configuration

#### What problem is this solving?
Resolves #3 

#### How should this be manually tested?
run (in the /theme folder)
> gatsby develop 
    
or
> yarn workspace theme develop 

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

#### How does this PR make you feel?
Happy gif

### CODE REVIEW

File: theme/public/page-data/dev-404-page/page-data.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"componentChunkName":"component---cache-dev-404-page-js","path":"/dev-404-page/","result":{"data":{"allSitePage":{"nodes":[{"path":"/"}]}},"pageContext":{}},"staticQueryHashes":[]}
```
Comment (thayannevls) [inline theme/public/page-data/dev-404-page/page-data.json:1]: Could you format this JSON?

Comment (JGuilhermeCoelho) [reply theme/public/page-data/dev-404-page/page-data.json]: Fixed #6

File: theme/src/pages/index.mdx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+title: Hello world!
+path: /page
+date: 2020-10-01
+author: Vtex
```
Comment (JRobsonJr) [inline theme/src/pages/index.mdx:5]:
```suggestion
author: VTEX
```
Be careful with this, or VTEX people will cancel you :rofl:

File: theme/public/page-data/index/page-data.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "componentChunkName": "component---src-pages-index-mdx",
+  "path": "/",
+  "result": {
+    "pageContext": {
+      "frontmatter": {
+        "title": "Hello world!",
+        "path": "/page",
+        "date": "2020-10-01T00:00:00.000Z",
+        "author": "Vtex"
```
Comment (JRobsonJr) [inline theme/public/page-data/index/page-data.json:10]:
```suggestion
        "author": "VTEX"
```

**Review by JRobsonJr — CHANGES_REQUESTED** [review]
Awesome! I left two small request changes :heart:

File: theme/src/pages/index.mdx (hunk 2 above)
Comment (JGuilhermeCoelho) [reply theme/src/pages/index.mdx:5]: I’m very sorry, It won't happen again

Comment (JRobsonJr) [reply theme/src/pages/index.mdx:5]: No problem, crisis averted :rofl: :heart:

**Review by LeandraOliveiraS — COMMENTED** [review]
Great job @JGuilhermeCoelho ! :smile:

**Review by LeandraOliveiraS — APPROVED** [review]
Great job @JGuilhermeCoelho! :smile:

**Review by JRobsonJr — APPROVED** [review]
Well done, Guilherme! :rocket:

**Pull request merged**